### DISCUSSION TITLE
How do you disable smart accounts?

### DISCUSSION THREAD

**Discussion by shunz19** [WalletConnect/walletconnect-monorepo#5534] [Developer Support]
When logging in with social account, the connected account defaults to a smart account. which throws error when calling provider.getSigner() for some reason.

When switching wallet address to a email one, calling provider.getSigner() works.

How do I disable smart accounts or atleast default to the non smart account one?

I'm using Ethers6 and React.

```
		"@reown/appkit": "^1.5.3",
		"@reown/appkit-adapter-ethers": "^1.5.3",
```

**Comment by arein** [comment]
Hi @shunz19 can you pls explain what you are trying to do and why you're calling `provider.getSigner()`? Also curious what are you building?

**Reply by shunz19** [reply]
Trying to sign a custom message for login verification.

It seems that they changed the default account to a "smart account" one on appkit social account when I asked this question. Currently the solution is to switch to a non smart account one and then you can sign messages with that. I opened an issue at reown appkit: https://github.com/reown-com/appkit/issues/3386

**Reply by arein** [reply]
gotcha. You're probably aware but in order to support logging in with Smart Contract wallets your verification logic needs to support EIP-1271/ERC-6492 signature verification.

Our example flows/logic are a good first step - we're shortly updating them though - https://viem.sh/docs/siwe/actions/verifySiweMessage is also a good pointer https://docs.reown.com/appkit/react/core/siwe#server-side

You can try logging in with the Social/Smart Account wallet here https://appkit-lab.reown.com/library/wagmi-siwe/

**Reply by arein** [reply]
This applies not just to the AppKit social wallet but also to all other smart contract wallets including safe and Coinbase

**Comment by irfanonk** [comment]
is it possible to disable smart account or make email account  default?

**Comment by evg-ch** [comment]
import { useAppKit, useAppKitAccount, useAppKitConnection } from '@reown/appkit/react'
import { useAccount } from 'wagmi'

const { address, addresses } = useAccount()
const { embeddedWalletInfo } = useAppKitAccount()
const { switchConnection } = useAppKitConnection({})

useEffect(() => {
    if (embeddedWalletInfo) {
      const isEoa = embeddedWalletInfo.accountType === 'eoa'
      if (!isEoa && addresses.length > 1) {
        const inactiveAddr = addresses.find((addr) => addr.toLowerCase() !== address.toLowerCase())
        switchConnection({
          address: inactiveAddr,
          connection: { accounts: addresses.map((addr) => ({ address: String(addr) })), connectorId: 'ID_AUTH' },
        })
      }
    }
  }, [address, embeddedWalletInfo])
